hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerImpl.java

```diff
@@ -784,6 +790,235 @@ public void testLookupKeyWithLocation() throws IOException {
         .getLocationList().get(0).getPipeline().getClosestNode());
   }
 
+  @Test
+  public void testListStatusWithTableCache() throws Exception {
+    // Inspired by TestOmMetadataManager#testListKeys
+    String prefixKeyInDB = "key-d";
+    String prefixKeyInCache = "key-c";
+
+    // Add a total of 100 key entries to DB and TableCache (50 entries each)
+    for (int i = 1; i <= 100; i++) {
+      if (i % 2 == 0) {  // Add to DB
+        TestOMRequestUtils.addKeyToTable(false,
+            VOLUME_NAME, BUCKET_NAME, prefixKeyInDB + i,
+            1000L, HddsProtos.ReplicationType.RATIS,
+            HddsProtos.ReplicationFactor.ONE, metadataManager);
+      } else {  // Add to TableCache
+        TestOMRequestUtils.addKeyToTableCache(
+            VOLUME_NAME, BUCKET_NAME, prefixKeyInCache + i,
+            HddsProtos.ReplicationType.RATIS, HddsProtos.ReplicationFactor.ONE,
+            metadataManager);
+      }
+    }
+
+    OmKeyArgs rootDirArgs = createKeyArgs("");
+    // Get entries in both TableCache and DB
+    List<OzoneFileStatus> fileStatuses =
+        keyManager.listStatus(rootDirArgs, true, "", 1000);
+    Assert.assertEquals(100, fileStatuses.size());
+
+    // Get entries with startKey=prefixKeyInDB
+    fileStatuses =
+        keyManager.listStatus(rootDirArgs, true, prefixKeyInDB, 1000);
+    Assert.assertEquals(50, fileStatuses.size());
+
+    // Get entries with startKey=prefixKeyInCache
+    fileStatuses =
+        keyManager.listStatus(rootDirArgs, true, prefixKeyInCache, 1000);
+    Assert.assertEquals(100, fileStatuses.size());
```

Comment:
This should be 50 right? As startKey is prefixKeyInCache. And from above we have created 50Keys with prefixKeyInDB and 50Keys with prefixKeyInCache
Assert.assertEquals(50, fileStatuses.size());

Reply:
@bharatviswa504 No it should be `100`. Because `prefixKeyInCache = "key-c"` and `prefixKeyInDB = "key-d"`, it will continue to list all entries beginning with `key-d`.

Reply:
Yaa got it.